src/commonMain/kotlin/dev/romainguy/kotlin/math/Matrix.kt

```diff
@@ -498,6 +502,117 @@ fun rotation(axis: Float3, angle: Float): Mat4 {
     )
 }
 
+/**
+ * Construct a matrix from Euler angles using YPR around ZYX respectively
+ *
+ * The Euler angles are applied in ZYX order.
+ * i.e: a vector is first rotated about X (roll) then Y (pitch) and then Z (yaw).
+ *
+ * @param d Per axis Euler angles in degrees
+ */
+fun euler(d: Float3): Mat4 {
+    val r = transform(d, ::radians)
+    return eulerZYX(r.z, r.y, r.x)
+}
+
+/**
+ * Construct a matrix from Euler angles using YPR around ZYX respectively
+ *
+ * The Euler angles are applied in ZYX order.
+ * i.e: a vector is first rotated about X (roll) then Y (pitch) and then Z (yaw).
+ *
+ * @param roll about X axis in radians
+ * @param pitch about Y axis in radians
+ * @param yaw about Z axis in radians
+ */
+fun eulerZYX(yaw: Float = 0.0f, pitch: Float = 0.0f, roll: Float = 0.0f): Mat4 {
+    val cy = cos(yaw)
+    val sy = sin(yaw)
+    val cp = cos(pitch)
+    val sp = sin(pitch)
+    val cr = cos(roll)
+    val sr = sin(roll)
+    return Mat4.of(cp * cy, cp * sy, -sp, 0.0f,
+            sp * sr * cy - cr * sy, sp * sr * sy + cr * cy, cp * sr, 0.0f,
+            sp * cr * cy + sr * sy, sp * cr * sy - sr * cy, cp * cr, 0.0f,
+            0.0f, 0.0f, 0.0f, 1.0f)
+}
+
+/**
+ * Construct a Quaternion Rotation Matrix following the Hamilton convention
+ *
+ * Assume the destination and local coordinate spaces are initially aligned, and the local
+ * coordinate space is then rotated counter-clockwise about a unit-length axis, k, by an angle,
+ * theta.
+ */
+fun quaternion(quaternion: Quaternion): Mat4 {
```

Comment:
You'll need a different name. `quaternion()` implies you are building a quaternion, not a matrix. You could call it `rotation()` if you want.

Reply:
Fixed https://github.com/romainguy/kotlin-math/pull/33/commits/e60495ba84a929cbda6ede258a067ad5baa72d13